pkg/services/baremetal/host/host.go

```diff
@@ -577,34 +580,56 @@ func (s *Service) actionRegistering() actionResult {
 		return s.recordActionFailure(infrav1.RegistrationError, errMsg)
 	}
 
-	if err := validateRootDevices(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints, s.scope.HetznerBareMetalHost.Spec.Status.HardwareDetails.Storage); err != nil {
+	if err := validateRootDeviceWwnsAreSubsetOfExistingWwns(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints,
+		s.scope.HetznerBareMetalHost.Spec.Status.HardwareDetails.Storage); err != nil {
 		conditions.MarkFalse(
 			s.scope.HetznerBareMetalHost,
 			infrav1.RootDeviceHintsValidatedCondition,
-			infrav1.StorageDeviceNotFoundReason,
+			infrav1.ValidationFailedReason,
 			clusterv1.ConditionSeverityError,
 			err.Error(),
 		)
 		return s.recordActionFailure(infrav1.RegistrationError, err.Error())
 	}
 
+	// Check RAID for the second time.
+	// See "tworaidchecks" for the other place.
+	msg = ""
+	if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid != 0 &&
+		len(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.Raid.WWN) < 2 {
+		msg = "Invalid HetznerBareMetalHost: spec.status.installImage.swraid is active. Use at least two WWNs in spec.rootDevideHints.raid.wwn."
+	} else if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid == 0 &&
+		s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.WWN == "" {
+		msg = "Invalid HetznerBareMetalHost: spec.status.installImage.swraid is not active. Use spec.rootDevideHints.wwn and leave raid.wwn empty."
+	}
+	if msg != "" {
+		s.scope.Logger.Info(msg)
```

Comment:
Add a comment that this triggers a FailureMessage on the HetznerBareMetalMachine and CAPI machine and will lead to this Machine to be deleted. 
Another machine will not take the same host anymore because the rootDeviceHints don't fit

Reply:
thanks!